pkg/http/listener.go

```diff
@@ -88,6 +89,19 @@ type httpListener struct {
 	errorLogFunc           func(error, string, ...interface{}) // function to be called on errors.
 }
 
+// isExpectedErr returns true if error is due to a network timeout or
+// io.EOF and false otherwise
+func isExpectedErr(err error) bool {
```

Comment:
1. I think this function as closure is more appropriate (you could remove unit test also)
2. you could name this function as 'ignoreErr()`